src/System.Numerics.Tensors/src/System/Numerics/Tensors/Tensor.cs

```diff
@@ -132,7 +132,7 @@ public static Tensor<T> CreateFromDiagonal<T>(Tensor<T> diagonal, int offset)
     /// <typeparam name="T">type contained within the Tensor.  Typically a value type such as int, double, float, etc.</typeparam>
     [DebuggerDisplay("{GetArrayString(false)}")]
     // When we cross-compile for frameworks that expose ICloneable this must implement ICloneable as well.
-    public abstract class Tensor<T> : IList, IList<T>, IReadOnlyList<T>, IStructuralComparable, IStructuralEquatable
+    public abstract class Tensor<T> : ITensor<T>, IList, IList<T>, IReadOnlyList<T>, IStructuralComparable, IStructuralEquatable
```

Comment:
With the `ITensor<T>` proposal, do you think we would still keep the abstract `Tensor<T>` class?